src/test/java/com/notably/logic/parser/suggestion/DeleteSuggestionCommandParserTest.java

```diff
@@ -91,374 +46,310 @@ public static void setUp() {
-        expectedSuggestions.add(cs2103Week2);
-        expectedSuggestions.add(cs2103Week3);
-        expectedSuggestions.add(cs2103Week1Lecture);
+        assertFalse(command.isPresent());
+    }
 
-        List<SuggestionItem> suggestions = model.getSuggestions();
+    @Test
+    public void parse_correctCmdWithoutPath_returnsOptionalEmpty() {
+        String userInput = "delete";
+        String path = "";
+
+        model.setInput(userInput);
+        Optional<? extends SuggestionCommand> command = deleteSuggestionCommandParser.parse(path);
 
-        // check display text
-        for (int i = 0; i < expectedSuggestions.size(); i++) {
-            SuggestionItem suggestion = suggestions.get(i);
-            SuggestionItem expectedSuggestion = expectedSuggestions.get(i);
-            assertEquals(expectedSuggestion.getProperty("displayText"), suggestion.getProperty("displayText"));
-        }
-
-        List<String> expectedInputs = new ArrayList<>();
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103Week1.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103Week2.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103Week3.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103Week1Lecture.getStringRepresentation());
-
-        for (int i = 0; i < expectedInputs.size(); i++) {
-            SuggestionItem suggestionItem = suggestions.get(i);
-            String expectedInput = expectedInputs.get(i);
-            suggestionItem.getAction().run();
-            String input = model.getInput();
-            assertEquals(expectedInput, input);
-        }
+        assertEquals(Optional.of(RESPONSE_MESSAGE),
+                model.responseTextProperty().getValue());
+
+        assertFalse(command.isPresent());
     }
 
     @Test
-    public void parse_correctAbsolutePathWithoutPrefix_returnsDeleteSuggestionCommand() {
-        String arg = toCs2103.getStringRepresentation();
-        model.setInput(COMMAND_WORD + " " + arg);
-        Optional<? extends SuggestionCommand> commandCorrectPath = deleteSuggestionCommandParser.parse(arg);
-        assertTrue(commandCorrectPath.get() instanceof DeleteSuggestionCommand);
+    public void parse_correctCmdcorrectAbsolutePathWithPrefix_returnsDeleteSuggestionCommand() {
+        String userInputWithoutPath = COMMAND_WORD + " " + PREFIX_TITLE + " ";
+        String userInput = userInputWithoutPath + toCs2103t.getStringRepresentation();
+        String arg = userInput.replace(COMMAND_WORD, "");
+
+        model.setInput(userInput);
+        Optional<? extends SuggestionCommand> command = deleteSuggestionCommandParser.parse(arg);
+        assertTrue(command.get() instanceof DeleteSuggestionCommand);
```

Comment:
Do you really need to test this? I'd think that testing the response message, the inputs, suggestions list, etc is already enough. Up to you!

Reply:
It's better if you assert sth like `assertTrue(command.isPresent())`